src/utils/multicall.ts

```diff
@@ -90,10 +81,56 @@ export const fetchDataUsingMulticall = async (
             }
           })
       } catch (e) {
-        return []
+        if (chunkSize === 1) {
+          // we already went down to the smallest chunk, log the error so we know which call failed and return
+          console.error(
+            `Failed to load balance for chainId ${chainId}`,
+            chunkedCalls[0]
+          )
+          return []
+        } else {
+          // try with chunk size of one to identify exactly the failed call
+          const smallestChunks = await chunkDataAndExecuteMulticall(
+            abiInterface,
+            multicallContract,
+            calls,
+            chainId,
+            1
```

Comment:
Correct me if I'm wrong but if a large chunk fails, this code will re-chunk that chunk with chunk size == 1 and re execute? 

If so, would it maybe be better to recurse by calling chunkSize / 2 ? This guarantees to reach the failing token in O(log(n)) calls and we won't waste bandwith / API call allowance / time in doing a ton of size 1 multicall calls.